### PR TITLE
MPMD cases

### PR DESCRIPTION
**Pull request by siddanib**
An initial **draft** of AMReX-MPMD cases. Includes two cases.

### CODE REVIEW

File: Docs/source/MPMD_Tutorials.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+Contents
+^^^^^^^^
+
+**Source_1** subfolder contains ``main_1.cpp`` which will be treated as the first application.
+Similarly, **Source_2** subfolder contains ``main_2.cpp`` that will be treated as the second application.
+
+Overview
+^^^^^^^^
+
+The domain in ``main_1.cpp`` is set to ``lo = {0, 0, 0}`` and ``hi = {31, 31, 31}``, while the domain in ``main_2.cpp`` is set to ``lo = {16, 16, 16}`` and ``hi = {31, 31, 31}``.
```
Comment (ajnonaka) [inline Docs/source/MPMD_Tutorials.rst:23]: You should mention that max_grid_size=16 in main_1.cpp and max_grid_size=8 in main_2.cpp, so each case has a different BoxArray.

File: Docs/source/MPMD_Tutorials.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+However, there exist scenarios where both applications deal with the same **BoxArray**.
+The current case presents such a scenario where the **BoxArray** is defined only in the ``main.cpp`` application, but this information is relayed to ``main.py`` application through the **MPMD::Copier**.
+
+**Please ensure that the same AMReX source code is used to compile both the applications.**
+
+pyAMReX compile
+^^^^^^^^^^^^^^^
+The compile process for pyAMReX is only briefly described here.
+Please refer to the `pyAMReX documentation <https://pyamrex.readthedocs.io/en/latest/install/cmake.html#>`_ for more details.
+It must be mentioned that **mpi4py** is an important dependency.
```
Comment (ajnonaka) [inline Docs/source/MPMD_Tutorials.rst:102]: add "make sure to conda install mpi4py when you are in the pyAMReX conda environment"

Comment (siddanib) [reply Docs/source/MPMD_Tutorials.rst]: Thanks for the inputs, Andy!

File: Docs/source/MPMD_Tutorials.rst [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+As mentioned earlier, the MPI ranks attributed to each application/code need to be continuous, i.e., MPI ranks 0-7 are for ``main.cpp`` and 8-11 are for ``main.py``.
+This may be default behaviour on several systems.
+Furthermore, the run process here assumes that the current working directory is ``ExampleCodes/MPMD/Case-2/``.
+
+.. code-block:: bash
+
+   # Running the MPMD process with 12 ranks
+   mpirun -np 8 ./main3d.gnu.DEBUG.MPI.ex : -np 4 python main.py
+
+Running on Perlmutter (NERSC)
```
Comment (ajnonaka) [inline Docs/source/MPMD_Tutorials.rst:137]:
```suggestion
Compiling and Running on Perlmutter (NERSC)
```

File: Docs/source/MPMD_Tutorials.rst [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+Overview
+^^^^^^^^
+
+In the previous case (Case-1) of MPMD each application has its own domain, and therefore, different **BoxArray**.
+However, there exist scenarios where both applications deal with the same **BoxArray**.
+The current case presents such a scenario where the **BoxArray** is defined only in the ``main.cpp`` application, but this information is relayed to ``main.py`` application through the **MPMD::Copier**.
+
+**Please ensure that the same AMReX source code is used to compile both the applications.**
+
```
Comment (ajnonaka) [inline Docs/source/MPMD_Tutorials.rst:97]:
```suggestion

Compiling and Running on a local system
```

File: Docs/source/MPMD_Tutorials.rst [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+Compiling and Running on a local system
+pyAMReX compile
+^^^^^^^^^^^^^^^
```
Comment (ajnonaka) [inline Docs/source/MPMD_Tutorials.rst:100]:
```suggestion
Compiling and Running on a local system
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: Docs/source/MPMD_Tutorials.rst [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+This may be default behaviour on several systems.
+Furthermore, the run process here assumes that the current working directory is ``ExampleCodes/MPMD/Case-2/``.
+
+.. code-block:: bash
+
+   # Running the MPMD process with 12 ranks
+   mpirun -np 8 ./main3d.gnu.DEBUG.MPI.ex : -np 4 python main.py
+
+Compiling and Running on Perlmutter (NERSC)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (ajnonaka) [inline Docs/source/MPMD_Tutorials.rst:138]:
```suggestion
Compiling and Running on Perlmutter (NERSC)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: Docs/source/MPMD_Tutorials.rst [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+   # Running the MPMD process with 12 ranks
+   mpirun -np 8 ./main3d.gnu.DEBUG.MPI.ex : -np 4 python main.py
+
+Compiling and Running on Perlmutter (NERSC)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Running this case on perlmutter involves creating a python virtual environment.
+pyAMReX must be compiled and installed into this virtual environment after its creation.
+Similar to the previous case, this case also has supporting scripts to run on CPUs and GPUs.
+
```
Comment (ajnonaka) [inline Docs/source/MPMD_Tutorials.rst:143]: make sure you are in Case-2/

**Review by ajnonaka — APPROVED** [review]
Looks great - go ahead and merge.

**Pull request merged**